pages/lists/[listId]/highlights.tsx

```diff
@@ -0,0 +1,267 @@
+  const supabase = createPagesServerClient(ctx);
+
+  const {
+    data: { session },
+  } = await supabase.auth.getSession();
+  const bearerToken = session ? session.access_token : "";
+
+  const { listId, repo } = ctx.params as { listId: string; repo?: string };
+  const limit = 10; // Can pull this from the querystring in the future
+  const [{ data, error: contributorListError }, { data: list, error }, { data: highlights, error: highlightError }] =
+    await Promise.all([
+      fetchApiData<PagedData<DBListContributor>>({
+        path: `lists/${listId}/contributors?limit=${limit}`,
+        bearerToken,
+        pathValidator: validateListPath,
+      }),
+      fetchApiData<DBList>({ path: `lists/${listId}`, bearerToken, pathValidator: validateListPath }),
+      fetchApiData<PagedData<DbHighlight>>({
+        path: `lists/${listId}/contributors/highlights?repo=${repo}`,
+        bearerToken,
+        pathValidator: validateListPath,
+      }),
+    ]);
+
+  if (error?.status === 404) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const userId = Number(session?.user.user_metadata.sub);
+
+  return {
+    props: {
+      list,
+      numberOfContributors: data?.meta.itemCount || 0,
+      isOwner: list && list.user_id === userId,
+      highlights: {
+        data: highlights?.data || [],
+        meta: highlights?.meta || {},
+      },
+    },
+  };
+};
+
+const rangeFilterOptions: { label: string; value: string }[] = [
+  {
+    label: "Past week",
+    value: "7",
+  },
+  {
+    label: "Past 30 days",
+    value: "30",
+  },
+  {
+    label: "Past 3 months",
+    value: "90",
+  },
+];
+
+const Highlights = ({ list, numberOfContributors, isOwner, highlights }: HighlightsPageProps) => {
+  const router = useRouter();
+  const repo = router.query.repo as string;
+
+  const { limit = 10, range = 30 } = router.query;
+  const topRef = useRef<HTMLDivElement>(null);
+
+  const { data: emojis } = useFetchAllEmojis();
+  const { data: taggedRepos } = useListHighlightsTaggedRepos(list?.id ?? "");
+
+  const [contributor, setContributor] = useState("");
+  const debouncedSearchTerm = useDebounceTerm(contributor, 300);
+
+  const { data, isLoading, meta } = useFetchListContributorsHighlights({
+    listId: list?.id ?? "",
+    initialData: highlights,
+    repo,
+    range: range as number,
+    contributor,
+    limit: limit as number,
+  });
+
+  function onSearch(searchTerm: string) {
+    if (!searchTerm || searchTerm.length >= 3) {
+      setContributor(searchTerm);
+    }
+  }
+
+  useEffect(() => {
+    onSearch(contributor);
+  }, [debouncedSearchTerm]);
+
+  return (
+    <ListPageLayout showRangeFilter={false} list={list} numberOfContributors={numberOfContributors} isOwner={isOwner}>
+      <div
+        ref={topRef}
+        className="container flex flex-col justify-start w-full gap-12 pt-4 mt-5 md:mt-0 md:items-start md:flex-row"
+      >
+        <div className={`sticky top-14 xl:flex hidden flex-none w-1/5`}>
```

Comment:
ux (blocking): smaller screens hide the search/filtering for highlighting. It should probably appear at the top for now, but we could iterate on better UX for this for small screens.

![CleanShot 2023-12-14 at 07 02 46](https://github.com/open-sauced/app/assets/833231/7a60ecfd-e03a-486f-905a-de618835f1d1)